packages/opencensus-exporter-stackdriver/src/stackdriver-stats-utils.ts

```diff
@@ -23,6 +24,55 @@ import {Distribution, LabelDescriptor, MetricDescriptor, MetricKind, MonitoredRe
 const OPENCENSUS_TASK = 'opencensus_task';
 const OPENCENSUS_TASK_DESCRIPTION = 'Opencensus task identifier';
 export const OPENCENSUS_TASK_VALUE_DEFAULT = generateDefaultTaskValue();
+const STACKDRIVER_PROJECT_ID_KEY = 'project_id';
+
+// Mappings for the well-known OC resources to applicable Stackdriver resources.
+const GCP_RESOURCE_MAPPING = getGcpResourceLabelsMappings();
+const K8S_RESOURCE_MAPPING = getK8sResourceLabelsMappings();
+const AWS_RESOURCE_MAPPING = getAwsResourceLabelsMappings();
+
+const GCP_GKE_CONTAINER = 'k8s_container';
+const GCP_GCE_INSTANCE = 'gce_instance';
+const AWS_EC2_INSTANCE = 'aws_ec2_instance';
+const GLOBAL = 'global';
+
+/* Return a self-configured StackDriver monitored resource. */
+export async function getDefaultResource(projectId: string):
+    Promise<MonitoredResource> {
+  let type;
+  const labels: Labels = {project_id: projectId};
+  let mappings: Labels = {};
+  const autoDetectedResource = await resource.detectResource();
+  switch (autoDetectedResource.type) {
```

Comment:
Optional: if you make a helper function that implements the logic of this `switch` then you can make an assignment like `const [type, mappings] = getTypeAndMappings(autoDetectedResource.type)`. This would avoid the `let` initializations.